doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+FORMAT: 1.0
+
+# REST service API for sharing configuration snippets
+
+This is a description of the API provided by the REST service that can be used to share, search and download configuration snippets.
+
+
+# API Description [/]
+
+## List API methods [GET]
+
+Returns an overview of available API methods with required parameters and a description.
+
++ Response 200 (text/html; charset=utf-8)
+
+
+
+# Authentication endpoint [/auth]
+
+## Authenticate [POST]
+
+The API will try to authenticate the user by the submitted credentials. In case of success a session token (JWT) will be returned, which can be used for further requests to the API.
```

Comment:
API Blueprints need an empty line between +Body and the example output, for example. So in order to have a more general layout, I decided to use 3 line breaks in front of each first level heading, 2 line breaks in front of each second heading, and so on.

JWT is what i decided to use, because it is very convenient to use in combination with SPA-frontends. And it actually doesn't make a difference for the user, he just has to know how to submit it.